### ISSUE TITLE
Routing of listSearchResults mainCategory

### ISSUE THREAD

**Issue by derBoogie** [jweiland-net/events2#367]
Hi,
after the upate to v8.0.1 I don't get the route configured for the category search.

My category selector looks like:
```
<ul>
	<li class="label">Categories</li>				
	<f:for each="{selectorData.categories.main}" as="category" iteration="catIterator">				
	<li><a href="{f:uri.page(pageUid: '{data.pid}', additionalParams: '{no-cache: 1, tx_events2_events: {search: {mainCategory: \'{category.uid}\'}, controller: \'Search\', action: \'listSearchResults\'}}')}">{category.title}</a></li>
	</f:for>
</ul>
```

The Links always looks like:
```
https://mywebsite.de/events?no-cache=1&tx_events2_events[action]=listSearchResults&tx_events2_events[controller]=Search&tx_events2_events[search][mainCategory]=27&cHash=8204fd43e5a1924e78e4e2f11990bda6
```

My config.yaml
```
...
  Events2ShowPlugin:
    type: Extbase
    limitToPages:
      - 113
    extension: Events2
    plugin: Show
    routes:
      -
        routePath: '/{date}/{event_title}'
        _controller: 'Day::show'
        _arguments:
          date: timestamp
          event_title: event
    requirements:
      date: '^\d{2,2}-\d{2,2}-\d{4,4}_\d{2,2}\.\d{2,2}$'
      event_title: '^[a-zA-Z0-9\-]+$'
    defaultController: 'Day::show'
    aspects:
      date:
        type: TimestampMapper
        format: d-m-Y_H.i
      event_title:
        type: PersistedPatternMapper
        tableName: tx_events2_domain_model_event
        routeFieldName: path_segment
        routeFieldPattern: '^(?P<path_segment>.+)$'
        routeFieldResult: '{path_segment}'
  Events2ListPlugin:
    type: Extbase
    limitToPages:
      - 111
    extension: Events2
    plugin: List
    routes:
      -
        routePath: '/seite-{page}'
        _controller: 'Day::list'
        _arguments:
          page: 'currentPage'
    requirements:
      page: \d+
    defaultController: 'Day::list'
    aspects:
      page:
        type: StaticRangeMapper
        start: '1'
        end: '99'
  Events2SearchResultsPlugin:
    type: Extbase
    limitToPages:
      - 111
    extension: Events2
    plugin: SearchResults
    routes:
      -
        routePath: '/kategorie/{category_title}'
        _controller: 'Search::listSearchResults'
        _arguments:
          category_title: search/mainCategory
      -
        routePath: '/kategorie/{category_title}/seite-{page}'
        _controller: 'Search::listSearchResults'
        _arguments:
          category_title: search/mainCategory
          page: 'currentPage'
    requirements:
      page: \d+
    defaultController: 'Search::listSearchResults'
    aspects:
      category_title:
        type: PersistedAliasMapper
        tableName: sys_category
        routeFieldName: slug
      page:
        type: StaticRangeMapper
        start: '1'
        end: '99'
...
```

What I'm doing wrong?

**Comment by froemken** [comment]
Hello @derBoogie 

In your additionalParams you have written `tx_events2_events`. This Plugin does not exists in events2 anymore. In you config.yaml section `Events2SearchResultsPlugin` you have written it correctly: `plugin: SearchResults`.
So, in your additionalParams you should write: `tx_events2_searchresults`.

Have a nice into this new week

Stefan

**Comment by derBoogie** [comment]
Hi @froemken

the speaking URLs are working right now with the adjustment of additionalParams to tx_events2_searchresults - thank's for that.

But filtering of the categories in the listing results is not working anymore.
I have only one page with the plugin "Liste aller Veranstaltungen" on it to show all events and the filtered events by category selection. Before the update, this worked well. Do I need to show the results of category selections on a separate page with the plugin "Suchergebnisse"?

Regards TJ

**Comment by derBoogie** [comment]
Hi @froemken 

I've integrated the search result listing on a separate page.
The filtered categories are now shown correctly on that page: https://www.wiesbaden-schelmengraben.de/veranstaltungen-und-termine/
This works for me.
Is the param "no-cache=1" really necessary?
I'm not using It right now, because it bothers me, just like the cHash parameter in the url.
The events are listed correctly, even without.

Regards TJ

**Comment by froemken** [comment]
Hello @derBoogie 

yes, you need two separate pages. Sorry, but now I have individual cache handling each plugin. list plugin is now cachable and searchresults plugin is configured as non-cached.

And yes, I hate no-cache, too. I have added a statement to FAQ section 1 or 2 years ago: https://docs.typo3.org/p/jweiland/events2/8.0/en-us/FAQ/Index.html#why-not-using-f-form-in-search-form

Maybe the problem is solved since all the individual plugins. Don't know.

Stefan

**Comment by derBoogie** [comment]
Hi @froemken 

I'm using the searchresults plugin for category selection with GET-requests (params in url) without the no-cache param and everything works fine, as you can see here: https://www.wiesbaden-schelmengraben.de/veranstaltungen-und-termine

Regards TJ

**Comment by froemken** [comment]
Hello @derBoogie 

Thank you for your feedback. I have set this ticket to "accepted". I think it's time to remove the no_cache=1 now. Since TYPO3 protocols all no_cache calls now, it will be a good step into the right direction.

I keep you informed about the process here.

Stefan

**Comment by derBoogie** [comment]
Hi @froemken 

that is good news!

Regards
TJ

**Comment by froemken** [comment]
Hi @derBoogie 

I have removed the no_cache part with that patch:
https://github.com/jweiland-net/events2/commit/66942900a85bef51281a9313d717282acca6cc65

It is solved in events2 8.1.1.

Have a nice start into the upcoming weekend

Stefan
